2016/proposals/singh-shubham.md

```diff
@@ -0,0 +1,191 @@
+This list can be filtered on the following criteria:
+  * instructor assigned.
+  * start date of training.
+  * mode of training (online or in person)
+  * location of the training.
+
+![HTML MOCKUP ALL_TRAINEE](https://cloud.githubusercontent.com/assets/15982349/13816909/3401c4a4-ebb6-11e5-9799-0eb21ccd5eea.png)
+4. *Trainee Home page* -
+After getting accepted for the training program, the trainees receive a randomly generated password, which they can use to log in to AMY, a trainee's home page is different from the AMY DASHBOARD, the AMY DASHBOARD is reserved for administrative purposes only. After logging in the trainees are directed to TRAINEE_HOME. This view will have a nav-bar - **HOME**, **ABOUT**, **NOTIFICATION** and **MESSAGES**.
+  * *HOME*  shows the trainees progress, through a progress bar indicating which stages have been completed and the ones left. In this view the trainee can also request a discussion session/checkout session by filling in a webform, with fields such as TOPIC, PREFERRED TIME, and so on.This view will also list down the general guidelines for instructor training, informing what is required in each stage of the training.
+
+  * *ABOUT* , this view shows the information entered by the trainee in their initial application, they can also edit this information, and change passwords here.
+  * *NOTIFICATION*,  this view allows the trainee to view all the comments/feedback from the discussion leaders and the lesson maintainers.In the format *comment: by whom: timestamp*.
+  * *MESSAGES*, In this view the trainee can talk to any of the Instructors/Discussion Leaders/Lesson Maintainers.We will be using [ django-postman](http://django-postman.readthedocs.org/en/latest/quickstart.html) for this.These are private messages.
+
+
+## Technical Details
+The Software stack for this project can be divided into:
+
+ * Backend  : [Django](https://www.djangoproject.com/)
+ * Database : [SQLite](https://www.sqlite.org/)
+ * Frontend : [Bootstrap](http://getbootstrap.com/),[JQuery](https://jquery.com/) for AJAX requests
+
+### Features and Model Schemes
+
+Trainee would be the main Model, maintaining most of the Flags for Progress and other information:
+
+```python
+class Trainee(models.Model):
+    '''Model for a Trainee '''
+    # listing just essential fields, more fields can and should be added when implementing
+    trainee = models.ForeignKey(Person,default=None)
+    badge   = models.ForeignKey(Badge)
+    group   = models.ForeignKey(Trainee_Group,default=None,null=True)
+    Attendance_Flag = models.BooleanField(default=False) # maintains attendance record
+    Discussion_Flag = models.BooleanField(default=False) # maintains PASS or FAIL
+    HomeWork_Flag = models.BooleanField(default=False) # maintains PASS or FAIL
+
+ ```
+
+Maintaining Group information:
+
+```python
+class Trainee_Group(models.Model):
+
+    instructor_assigned= models.ForeignKey(Person)
```

Comment:
Mm - doesn't it make more sense to link a trainee group to an event, because events may have multiple instructors?